### PR TITLE
docs: add hints about how to use snapshot version of `generator-cli`

### PR DESCRIPTION
**Pull request by joscha**
This adds documentation on how to use a locally built JAR and/or a snapshot version with the npm-based generator package.

@macjohnny as promised 🎉 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -445,6 +445,44 @@ Or install it as dev-dependency:
...
+```
+and then:
+```json
+{
+    "$schema": "./node_modules/@openapitools/openapi-generator-cli/config.schema.json",
+    "spaces": 2,
+    "generator-cli": {
+        "version": "my-local-snapshot",
+    },
+    // ...
```
Comment (joscha) [inline README.md:465]:
```suggestion
    }
```

**Comment by macjohnny** [comment]
@joscha thanks!

**Comment by macjohnny** [comment]
@joscha would you want to add this change to the readme in https://github.com/OpenAPITools/openapi-generator-cli as well?

**Comment by joscha** [comment]
> @joscha would you want to add this change to the readme in https://github.com/OpenAPITools/openapi-generator-cli as well?

Ah yes, good call! I can also move it, so it's not duplicated? Although, I did search in this readme before, hence why I added it here. Let me know!

**Comment by joscha** [comment]
Changes here: https://github.com/OpenAPITools/openapi-generator-cli/pull/794. Will change the content in this repository to a link once the https://github.com/OpenAPITools/openapi-generator-cli/pull/794 has been merged.

**Comment by wing328** [comment]
@joscha when you've time, can you please PM me via Slack? thanks.

https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g

**Comment by joscha** [comment]
> > @joscha would you want to add this change to the readme in https://github.com/OpenAPITools/openapi-generator-cli as well?
> 
> Ah yes, good call!
> 
> ~I can also move it, so it's not duplicated? Although, I did search in this readme before, hence why I added it here. Let me know!~
> 
> Will link it.

Follow-up is here: https://github.com/OpenAPITools/openapi-generator/pull/19544

**Pull request merged**